packages/ember-data/lib/adapters/rest-adapter.js

```diff
@@ -774,11 +779,11 @@ export default Adapter.extend(BuildURLMixin, {
     @param {Object} options
     @return {Promise} promise
   */
-  ajax: function(url, type, options) {
+  ajax: function(request) {
```

Comment:
#2582 shows that `ajax` - though marked as `@private` - is used to make ajax calls within overwritten adapters. Those calls will break with this refactor. Should the old behavior be supported and a deprecation warning be added?